### PR TITLE
fix(kubernetes_cmd_runner): make it threadsafe

### PR DESCRIPTION
**Pull request by dkropachev**
https://trello.com/c/sY1HNzYO/2367-investigate-why-k8s-remoter-is-unstable

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

**Review by bentsi — APPROVED** [review]

File: sdcm/remote/kubernetes_cmd_runner.py [hunk 1]
```diff
@@ -179,3 +188,9 @@ def send_files(self, src, dst, delete_dst=False, preserve_symlinks=False, verbos
         KubernetesOps.copy_file(self, src, f"{self.namespace}/{self.pod}:{dst}",
                                 container=self.container, timeout=timeout)
         return True
+
+    def stop(self):
+        pass
```
Comment (enaydanov) [inline sdcm/remote/kubernetes_cmd_runner.py:193]: Need to stop connection here.

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: sdcm/remote/kubernetes_cmd_runner.py [hunk 2]
```diff
@@ -179,3 +188,9 @@ def send_files(self, src, dst, delete_dst=False, preserve_symlinks=False, verbos
         KubernetesOps.copy_file(self, src, f"{self.namespace}/{self.pod}:{dst}",
                                 container=self.container, timeout=timeout)
         return True
+
+    def stop(self):
```
Comment (bentsi) [inline sdcm/remote/kubernetes_cmd_runner.py:192]: close connection

File: sdcm/remote/kubernetes_cmd_runner.py [hunk 3]
```diff
@@ -61,16 +61,20 @@ def close_proc_stdin(self) -> None:
...
+                    container=self.context.config.k8s_container,
+                    namespace=self.context.config.k8s_namespace,
+                    command=[shell, "-c", command],
+                    stderr=True,
+                    stdin=True,
+                    stdout=True,
+                    tty=False,
+                    _preload_content=False)
+            except k8s.client.rest.ApiException as exc:
+                raise ConnectionError(str(exc))
```
Comment (enaydanov) [inline sdcm/remote/kubernetes_cmd_runner.py:77]: Please, change it to `raise ... from None`

File: sdcm/remote/kubernetes_cmd_runner.py [hunk 4]
```diff
@@ -151,8 +155,13 @@ def _run():
...
             try:
-                connection = self._create_connection() if new_session else self.connection
-                res = connection.run(command=cmd, warn=ignore_status, hide=True, watchers=watchers, timeout=timeout)
+                res = self._create_connection().run(
+                    command=cmd,
+                    warn=ignore_status,
+                    hide=True,
+                    watchers=watchers,
+                    timeout=timeout
+                )
```
Comment (enaydanov) [inline sdcm/remote/kubernetes_cmd_runner.py:164]: Put it on the same line (or add comma)

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: sdcm/remote/kubernetes_cmd_runner.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/remote/kubernetes_cmd_runner.py:192]: Added comment there

File: sdcm/remote/kubernetes_cmd_runner.py (hunk 3 above)
Comment (dkropachev) [reply sdcm/remote/kubernetes_cmd_runner.py:77]: Done

File: sdcm/remote/kubernetes_cmd_runner.py (hunk 4 above)
Comment (dkropachev) [reply sdcm/remote/kubernetes_cmd_runner.py:164]: done

File: sdcm/remote/kubernetes_cmd_runner.py (hunk 1 above)
Comment (dkropachev) [reply sdcm/remote/kubernetes_cmd_runner.py:193]: Added comment there

**Pull request merged**